pmd-ui/src/main/java/net/sourceforge/pmd/util/fxdesigner/MainDesignerController.java

```diff
@@ -410,24 +309,13 @@ public void setRecentFiles(String files) {
 
     @PersistentProperty
     public boolean isMaximized() {
-        return designerRoot.getMainStage().isMaximized();
+        return getMainStage().isMaximized();
     }
 
 
     public void setMaximized(boolean b) {
-        designerRoot.getMainStage().setMaximized(!b); // trigger change listener anyway
-        designerRoot.getMainStage().setMaximized(b);
-    }
-
-
-    @PersistentProperty
-    public boolean isBottomTabExpanded() {
```

Comment:
I've tested it, it appears to be working. Afterwards, the properties is removed from the designer.xml.

Reply:
Yes I'm pretty sure the persistence component handles it gracefully. Properties have been moved from a controller to another on several occasions already. The only bug I've encountered with it since it was introduced is its handling of null values, which is fixed in a topic branch for the next version.